### PR TITLE
CCR: mention how to enable soft deletes in error message when not enabled

### PR DESCRIPTION
**Pull request by bleskes**

### CODE REVIEW

**Review by martijnvg — APPROVED** [review]
LGTM - Do think tests need to be adjusted because of this change. (`LocalFollowIndexIT#testDoNotCreateFollowerIfLeaderDoesNotHaveSoftDeletes(...)` and `TransportResumeFollowActionTests#testValidation(...)`)

**Comment by bleskes** [comment]
Thanks @martijnvg, I was optimistically relying on CI to be bubble up tests that rely on this string (if at all). Thanks for the heads up.

File: x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/TransportPutFollowAction.java [hunk 1]
```diff
@@ -124,7 +124,8 @@ private void createFollowerIndex(
         }
         if (leaderIndexMetaData.getSettings().getAsBoolean(IndexSettings.INDEX_SOFT_DELETES_SETTING.getKey(), false) == false) {
             listener.onFailure(
-                new IllegalArgumentException("leader index [" + request.getLeaderIndex() + "] does not have soft deletes enabled"));
+                new IllegalArgumentException("leader index [" + request.getLeaderIndex() + "] does not have soft deletes enabled." +
+                    "soft deletes must be enabled when the index is created by setting index.soft_deletes.enabled to true"));
```
Comment (jasontedor) [inline x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/TransportPutFollowAction.java]: Please use `IndexSettings.INDEX_SOFT_DELETES_SETTING.getKey()` so we have fewer places we have to search the code base for the hard string for, if ever we need to.

**Review by jasontedor — CHANGES_REQUESTED** [review]
🙏

**Review by jasontedor — APPROVED** [review]

**Comment by bleskes** [comment]
check this please

**Comment by bleskes** [comment]
@martijnvg @jasontedor FYI - I rebased this against 6.x . With the merge of #36141 which enables soft deletes as a default, I don't think this makes sense for master aymore.

**Pull request merged**